### PR TITLE
New design: package show page + versions page

### PR DESCRIPTION
**Pull request by isoos**
I've tried to stick to the following refactor logic:
- Reuse as much as possible, don't duplicate code. I've extracted value calculation, old method goes to the front, new method afterward (that way the diff is reasonable).
- I've commented a few lines with //experimental_design: obsolete or new. These are to make it easier to clean up after the merge.

### CODE REVIEW

**Comment by isoos** [comment]
Shall I put this into multiple pieces? E.g. tag rendering can be easily extracted and it is being shared between all of the new pages.

File: app/test/frontend/handlers_test_utils.dart [hunk 1]
```diff
@@ -202,12 +202,31 @@ class TemplateMock implements TemplateService {
   }
 
   @override
+  String renderNewPkgShowPage(
```
Comment (mkustermann) [inline app/test/frontend/handlers_test_utils.dart]: Maybe we should be consistent with what we use `new` or `experimental` or `v2`. I would just go for `v2` and  only use `experimental` in the URL, so users know that this is not stable yet.

File: app/views/v2/pkg/show.mustache [hunk 2]
```diff
@@ -0,0 +1,147 @@
+{{! Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
```
Comment (mkustermann) [inline app/views/v2/pkg/show.mustache]: We should consider adding golden files for the v2 templates as well.

File: app/lib/frontend/templates.dart [hunk 3]
```diff
@@ -325,12 +379,43 @@ class TemplateService {
...
       'version_count': '$totalNumberOfVersions',
+      'icons': LogoUrls.versionsTableIcons, // experimental_design: new
     };
-    return _renderPage(
-      'pkg/show',
-      values,
+    return values;
+  }
+
+  /// Renders the `views/private_keys/show.mustache` template.
```
Comment (mkustermann) [inline app/lib/frontend/templates.dart]: This comment seems wrong.

File: app/lib/frontend/handlers.dart [hunk 4]
```diff
@@ -465,6 +547,13 @@ Future<shelf.Response> packageVersionHandlerHtml(
   return htmlResponse(cachedPage);
 }
 
+/// Handles requests for /experimental/packages/<package>/versions/<version>
+Future<shelf.Response> experimentalPackageVersionHandlerHtml(
+    shelf.Request request, String packageName, String versionName) {
+  return _packageVersionHandlerHtml(request, packageName, versionName,
+      templateService.renderNewPkgShowPage, false);
```
Comment (mkustermann) [inline app/lib/frontend/handlers.dart]: I would prefer to have a `backed.uiPackageCacheV2` which can be used for caching in the new v2 handlers and invalidation is invalidating `backed.uiPackageCache` and `backend.uiPackageCacheV2`.

Then we don't need all these booleans.

File: app/views/v2/pkg/show.mustache (hunk 2 above)
Comment (isoos) [reply app/views/v2/pkg/show.mustache]: added

File: app/test/frontend/handlers_test_utils.dart (hunk 1 above)
Comment (isoos) [reply app/test/frontend/handlers_test_utils.dart]: Renamed them with a V2 postfix.

File: app/lib/frontend/templates.dart (hunk 3 above)
Comment (isoos) [reply app/lib/frontend/templates.dart]: Yeah, the original of the copy-paste was wrong too, fixed both.

File: app/lib/frontend/handlers.dart (hunk 4 above)
Comment (isoos) [reply app/lib/frontend/handlers.dart]: The flag in https://github.com/dart-lang/pub-dartlang-dart/pull/415 will substitute the bool here.

**Pull request merged**